### PR TITLE
[AIP-11] Allow round creation via Registry

### PR DESCRIPTION
**Pull request by thelostone-mc**
Based off the discussion off : https://github.com/orgs/allo-protocol/discussions/8
PR : https://github.com/allo-protocol/aips/pull/19

### CODE REVIEW

**Review by KurtMerbeth — APPROVED** [review]

**Comment by gravityblast** [comment]
> Updating the project struct to include projectMetadata provides a more comprehensive representation of projects

The project struct already has the metadata pointer, so do you mean just rename it from metadata to projectMetadata? 
I think it would be redundant since it's already inside the Project struct, so I would prefer `project.metadata` than `project.project_metadata`

**Comment by gravityblast** [comment]
> - Update the metadata event to emit the metadata type.

The even already has the metadata type in the pointer 
```
    event MetadataUpdated(uint256 indexed projectID, MetaPtr metaPtr);
```

**Comment by KurtMerbeth** [comment]
@gravityblast 
The proposed change involves removing the Program contracts and introducing a second metapointer to the Project Struct. This means that a Project will now have both a project-metapointer, as before, and an additional program-metapointer.

Since a second metapointer is introduced, a new function for the second metapointer is added. To simplify the implementation, a single event is used when updating a metapointer. To distinguish between the two metapointers, a type (enum) is introduced, indicating whether the updated metapointer corresponds to the Project or Program metapointer.

**Comment by gravityblast** [comment]
> Since a second metapointer is introduced, a new function for the second metapointer is added. To simplify the implementation, a single event is used when updating a metapointer. To distinguish between the two metapointers, a type (enum) is introduced, indicating whether the updated metapointer corresponds to the Project or Program metapointer.

@KurtMerbeth ok so we should update the document because now it says:

`- Update the project struct to include projectMetadata for a more comprehensive representation of projects.`

`Introduce the updateProjectMetadata function in the Registry contract to allow updates to project metadata. `

so maybe they need to be renamed to "update *Program* metadata"?
or do we need to change the text to add a parameter to the already existing methods?

File: AIPS/aip-11.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        uint256 id;
+        MetaPtr projectMetadata;
+        MetaPtr programMetadata;
+    }
+```
+
+- Add the following function to the Registry contract:
+
+```solidity
+    function updateProjectMetadata(
```
Comment (thelostone-mc) [inline AIPS/aip-11.md:77]:
```suggestion
    function updateProgramMetadata(
```

File: AIPS/aip-11.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+## Abstract
+
+This AIP proposes several changes to the Registry, RoundFactory, and AlloSettings contracts in order to enhance functionality, improve security, and streamline contract interactions in the project ecosystem.
+
+## Motivation
+
+The motivation behind these proposed changes is to address certain limitations and inefficiencies in the existing contracts. By renaming the ProjectRegistry contract to simply Registry, we aim to simplify the naming convention and make it more intuitive for users. The addition of the createRound function in the Registry contract allows projects to create rounds using a unique ID, which is constructed based on the chain ID, registry address, and project ID. This improves the management and identification of rounds within the system.
+
+Furthermore, by purging the ProgramFactory and ProgramImplementation contracts, we eliminate redundant functionality and simplify the contract structure. Updating the project struct to include projectMetadata provides a more comprehensive representation of projects, while the introduction of the updateProjectMetadata function enables easy updates to the project metadata.
```
Comment (thelostone-mc) [inline AIPS/aip-11.md:17]:
```suggestion
Furthermore, by purging the ProgramFactory and ProgramImplementation contracts, we eliminate redundant functionality and simplify the contract structure. Updating the project struct to include programMetadata provides a more comprehensive representation of projects, while the introduction of the updateProgramMetadata function enables easy updates to the program metadata.
```

**Comment by thelostone-mc** [comment]
@gravityblast You are right. I had confused the name while writing the AIP. 
We are adding a new updateProgramMetadata. Like you mentioned we already have updateProjectMetadata.
Updated the AIP

**Comment by gravityblast** [comment]
nice thank you @thelostone-mc !  ❤️

**Pull request merged**